app/src/main/java/org/wikipedia/settings/RemoteConfigRefreshTask.kt

```diff
@@ -32,6 +35,10 @@ class RemoteConfigRefreshTask : RecurringTask() {
             } finally {
                 response?.closeQuietly()
             }
+
+            val userInfo = ServiceFactory.get(WikipediaApp.instance.wikiSite).getUserInfo()
+            // This clumsy comparison is necessary because the field is an integer value when enabled, but a string when disabled.
+            Prefs.donationBannerOptIn = userInfo.query?.userInfo?.options?.fundraisingOptIn?.jsonPrimitive.toString() == "1"
```

Comment:
Just curious, if one of the values is `null`, will the optIn preference become `false` because `null` or `""` is not equal to `1` by default?

Reply:
Yeah, fair point. We do want the default for this preference to lean towards `true`, so I modified this check to look for an explicit empty string `""`, which would definitely indicate `false`.

Reply:
Thanks. I checked the latest commit you made, if the `userInfo.query?.userInfo?.options?.fundraisingOptIn?` output `null`, and then `toString().orEmpty().replace("\"", "")` will convert it to an empty string. Is this something expected?

Reply:
It is not! And... fixed.